### PR TITLE
Id porten api

### PR DESCRIPTION
**Pull request by simen-rekkedal**
Create a new endpoint for IdPorten for the ClientId 

## Description

The ClientId is maintained by IdPorten and has a one to one mapping to the System User Integration Ids,
for a given Consumer ( Party ) Id.

## Related Issue(s)
- Create API to search for systemuser from Maskinporten [#337](https://github.com/Altinn/altinn-authentication/issues/337)

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/ISystemUserService.cs [hunk 1]
```diff
@@ -13,4 +13,5 @@ public interface ISystemUserService
     Task<bool> ChangeSystemUserTitle(string newTitle, Guid id, CancellationToken cancellationToken = default);
     Task<bool> ChangeSystemUserDescription(string newDescr, Guid id, CancellationToken cancellationToken = default);
     Task<bool> ChangeSystemUserProduct(string selectedSystemType, Guid id, CancellationToken cancellationToken = default);
+    Task<SystemUserDTO?> CheckIfPartyHasIntegration(string clientId, string partyId, CancellationToken cancellationToken);
```
Comment (mgunnerud) [inline bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/ISystemUserService.cs:16]:
```suggestion
    Task<SystemUserDTO?> CheckIfPartyHasIntegration(string clientId, string partyId, CancellationToken cancellationToken = default);
```
?

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/SystemUserClient.cs [hunk 2]
```diff
@@ -88,7 +85,7 @@ public Task<bool> ChangeSystemUserRealDescription(string newDescr, Guid id, Canc
 
     public async Task<List<SystemUserReal>> GetSystemUserRealsForChosenUser(int id, CancellationToken cancellationToken = default)
     {
-        List<SystemUserReal> list = new();
+        List<SystemUserReal> list = [];
```
Comment (mgunnerud) [inline bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/SystemUserClient.cs:88]: Blir ikke dette å assigne en array til en list? Kanskje bedre med `new()` ?

Comment (simen-rekkedal) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/SystemUserClient.cs:88]: Nei, det er ny notasjon i dotnet 8. Det blir det samme som new().

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/SystemUserController.cs [hunk 3]
```diff
@@ -56,6 +69,25 @@ public async Task<ActionResult> GetSystemUserDetailsById(Guid guid, Cancellation
...
+    /// PartyId is the firste entry in the path.
+    /// ClientId is the second entry in the path.
+    /// </summary>
+    /// <param name="clientId">The unique id maintained by IdPorten uniquely tying their clients to the Integration Ids we maintain, 1 to 1</param>
+    /// <param name="cancellationToken">Cancellationtoken</param>
+    /// <param name="consumerId">The legal number (Orgno) of the enduser owning the System User Integration</param>
+    /// <returns>The SystemUserIntegration model API DTO</returns>
+    //[Authorize]
+    [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+    [HttpGet("get-consumers-integration-by-clientId/{consumerId}/{clientId}")]
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/SystemUserController.cs]: Ja, men dette er et spesielt endepunkt som kun skal brukes av maskinporten, det inngår ikke i den "vanlige crud" delen av APIet

**Review by mgunnerud — APPROVED** [review]

**Pull request merged**